lib/notifications/display.dart

```diff
@@ -92,7 +93,29 @@ class NotificationDisplayManager {
   static Future<void> _onMessageFcmMessage(MessageFcmMessage data, Map<String, dynamic> dataJson) async {
     assert(debugLog('notif message content: ${data.content}'));
     final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
-    final title = switch (data.recipient) {
+    final groupKey = _groupKey(data);
+    final conversationKey = _conversationKey(data, groupKey);
+
+    final oldMessagingStyle = await ZulipBinding.instance.androidNotificationHost
+      .getActiveNotificationMessagingStyleByTag(conversationKey);
+
+    final MessagingStyle messagingStyle = oldMessagingStyle != null
+      ? MessagingStyle(
+          user: oldMessagingStyle.user,
+          messages: oldMessagingStyle.messages?.toList() ?? [], // Clone a fixed-length list
+          isGroupConversation: oldMessagingStyle.isGroupConversation)
```

Comment:
Is there a reason to create a new `MessagingStyle` based on `oldMessagingStyle` (but with `conversationTitle` omitted)? What if we just used `oldMessagingStyle`?

I did get a test failure when I tried this, but it's pretty opaque to me:

<details>

```
00:18 +770 ~6 -1: /Users/chrisbobbe/dev/zulip-flutter/test/notifications/display_test.dart: NotificationDisplayManager show multiple stream messages [E]
  Expected: a List<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})> that:
    has length that:
      equals <6>
    contains, in order: [Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void,
    Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void,
    Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void,
    Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void,
    Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void,
    Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void]
  Actual: [(autoCancel: true, channelId: messages-1, color: 4284781310, contentIntent: Instance of 'PendingIntent', contentText: null, contentTitle: null, extras: null, groupKey: https://chat.example/|1084, id: 1266374773, inboxStyle: null, isGroupSummary: null, messagingStyle: Instance of 'MessagingStyle', number: 1, smallIconResourceName: zulip_notification, tag: https://chat.example/|1084|stream:1:example topic),
  (autoCancel: true, channelId: messages-1, color: 4284781310, contentIntent: null, contentText: null, contentTitle: null, extras: null, groupKey: https://chat.example/|1084, id: 712478168, inboxStyle: Instance of 'InboxStyle', isGroupSummary: true, messagingStyle: null, number: null, smallIconResourceName: zulip_notification, tag: https://chat.example/|1084),
  (autoCancel: true, channelId: messages-1, color: 4284781310, contentIntent: Instance of 'PendingIntent', contentText: null, contentTitle: null, extras: null, groupKey: https://chat.example/|1084, id: 1266374773, inboxStyle: null, isGroupSummary: null, messagingStyle: Instance of 'MessagingStyle', number: 2, smallIconResourceName: zulip_notification, tag: https://chat.example/|1084|stream:1:example topic),
  (autoCancel: true, channelId: messages-1, color: 4284781310, contentIntent: null, contentText: null, contentTitle: null, extras: null, groupKey: https://chat.example/|1084, id: 712478168, inboxStyle: Instance of 'InboxStyle', isGroupSummary: true, messagingStyle: null, number: null, smallIconResourceName: zulip_notification, tag: https://chat.example/|1084),
  (autoCancel: true, channelId: messages-1, color: 4284781310, contentIntent: Instance of 'PendingIntent', contentText: null, contentTitle: null, extras: null, groupKey: https://chat.example/|1084, id: 1266374773, inboxStyle: null, isGroupSummary: null, messagingStyle: Instance of 'MessagingStyle', number: 3, smallIconResourceName: zulip_notification, tag: https://chat.example/|1084|stream:1:example topic),
  (autoCancel: true, channelId: messages-1, color: 4284781310, contentIntent: null, contentText: null, contentTitle: null, extras: null, groupKey: https://chat.example/|1084, id: 712478168, inboxStyle: Instance of 'InboxStyle', isGroupSummary: true, messagingStyle: null, number: null, smallIconResourceName: zulip_notification, tag: https://chat.example/|1084)]
  Which: did not have an element matching the expectation at index 0 <Closure: (Subject<({bool? autoCancel, String channelId, int? color, PendingIntent? contentIntent, String? contentText, String? contentTitle, Map<String?, String?>? extras, String? groupKey, int id, InboxStyle? inboxStyle, bool? isGroupSummary, MessagingStyle? messagingStyle, int? number, String? smallIconResourceName, String? tag})>) => void>
  package:checks/src/checks.dart 85:9                check.<fn>
  package:checks/src/checks.dart 708:12              _TestContext.expect
  package:checks/src/extensions/iterable.dart 93:13  IterableChecks.containsInOrder
  test/notifications/display_test.dart 188:11        main.<fn>.checkNotification
  test/notifications/display_test.dart 205:7         main.<fn>.checkNotifications
  test/notifications/display_test.dart 241:13        main.<fn>.<fn>.<fn>
  

To run this test again: /Users/chrisbobbe/.local/lib/flutter/bin/cache/dart-sdk/bin/dart test /Users/chrisbobbe/dev/zulip-flutter/test/notifications/display_test.dart -p vm --plain-name 'NotificationDisplayManager show multiple stream messages'
00:43 +1192 ~168 -1: Some tests failed.     
```

</details>